ui/src/auth/auth0OAuth.tsx

```diff
@@ -0,0 +1,163 @@
+import { Auth0Provider, useAuth0 } from "@auth0/auth0-react";
+import Box from "@mui/material/Box";
+import Button from "@mui/material/Button";
+import { loginAccessType } from "apiContext";
+import { ErrorList } from "components/errorPage";
+import { getEnvironment } from "environment";
+import verilyImage from "images/verily.png";
+import GridLayout from "layout/gridLayout";
+import React, { useCallback, useEffect } from "react";
+import { useNavigate } from "react-router";
+import { Outlet, Path, redirect, useLocation } from "react-router-dom";
+
+export function isAuth0Enabled(): boolean {
+  return !!getEnvironment().REACT_APP_AUTH0_DOMAIN;
+}
+
+export function Auth0ProviderWithOutlet() {
+  const env = getEnvironment();
+  const navigate = useNavigate();
+
+  if (!isAuth0Enabled) {
+    return null;
+  }
+
+  return (
+    <Auth0Provider
+      domain={env.REACT_APP_AUTH0_DOMAIN}
+      clientId={env.REACT_APP_AUTH0_CLIENT_ID}
+      cacheLocation="localstorage"
+      onRedirectCallback={(appState) =>
+        navigate(appState?.returnTo || "/", { replace: true })
+      }
+      authorizationParams={{
+        scope: "profile email",
+        redirect_uri: window.location.origin,
+      }}
+    >
+      <Outlet />
+    </Auth0Provider>
+  );
+}
+
+function returnToPreviousPage(shouldNavigate: boolean, reason: string) {
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (shouldNavigate) {
+      console.info("Return to previous page: " + reason);
+      const from: Path | undefined = location.state?.from;
+      navigate(from || "/", { replace: true });
+    }
+  }, [location, navigate]);
+}
+
+export function LoginPage() {
```

Comment:
logout --> From discussion - we will keep the logout option, at least needed for testing. We will not need a login / logout page especially since tanagra iframe should use the primary app's functionalities. 

login --> I am having trouble with replacing the login page with a direct call to Auth0. Issue: it leads to an infinite loop of page re-render that includes auth check calls. I might be wrong in my assumption: replacing the explicit UI button with implicit call might not work the way we intend it to. 

given that both login and logout pages are for the sample app (with verily image), and there are plans to update the VWB-tanagra integration in the near future I am inclined to keep them for now. Trying to avoid extra time spent debugging and polishing code that would likely be removed